python_modules/libraries/dagster-databricks/dagster_databricks/databricks_step_main.py

```diff
@@ -0,0 +1,118 @@
+'''The main entrypoint for solids executed using the DatabricksPySparkStepLauncher.
+
+This script is launched on Databricks using a `spark_python_task` and is passed the following
+parameters:
+
+- the DBFS path to the pickled `step_run_ref` file
+- the DBFS path to the zipped pipeline
+- paths to any other zipped packages which have been uploaded to DBFS.
+'''
+
+import os
+import pickle
+import site
+import sys
+import tempfile
+import zipfile
+
+from dagster.core.execution.plan.external_step import PICKLED_EVENTS_FILE_NAME, run_step_from_ref
+from dagster.serdes import serialize_value
+
+
+# This won't be set in Databricks but is needed to be non-None for the
+# Dagster step to run.
+if 'DATABRICKS_TOKEN' not in os.environ:
+    os.environ['DATABRICKS_TOKEN'] = ''
+
+
+def setup_s3_storage(scope, credentials):
+    '''Obtain AWS credentials from Databricks secrets and export so both Spark and boto can use them.
+    '''
+    access_key = dbutils.secrets.get(scope=scope, key=credentials['access_key_key'])
+    secret_key = dbutils.secrets.get(scope=scope, key=credentials['secret_key_key'])
+
+    # Spark APIs will use this.
+    # See https://docs.databricks.com/data/data-sources/aws/amazon-s3.html#alternative-1-set-aws-keys-in-the-spark-context.
+    sc._jsc.hadoopConfiguration().set('fs.s3n.awsAccessKeyId', access_key)
```

Comment:
Where is sc coming from here?

Reply:
It's a global defined in the Databricks runtime pointing to the Spark context. I've added a comment explaining it and the `dbutils` global.